packages/runner/cypress/plugins/snapshot/snapshotCommand.js

```diff
@@ -0,0 +1,447 @@
+const _ = require('lodash')
+// if we're in Cypress, we'll need to swap this with Cypress.sinon later
+let sinon = require('sinon')
+const Debug = require('debug')
+const chalk = require('chalk')
+const stripAnsi = require('strip-ansi')
+const { stripIndent } = require('common-tags')
+const { printVar, stringifyShort, isObject, addPluginButton, fmt, typeColors } = require('./snapshotUtils')
+
+const debug = Debug('plugin:snapshot')
+
+/**
+ * prints nice assertion error in command log with modified error message
+ */
+function throwErr (e, message, exp, ctx) {
+  try {
+    ctx.assert(false, message, 'sdf', exp, e.act, true)
+  } catch (err) {
+    err.message += `\n\n**- expected  + actual:**\n${e.message}`
+    throw err
+  }
+}
+
+function getMatchDeepMessage ({ act, exp }) {
+  return `Expected **${chai.util.objDisplay(act)}** to deep match: **${chai.util.objDisplay(exp)}**`
+}
+
+function saveSnapshot (ctx, exactSpecName, file, exp, act) {
+  const message = !exp ? 'new snapshot saved' : 'snapshot updated'
+
+  ctx.assert(true, `📸 ${message}: **${exactSpecName}**`, '', exp, act)
+
+  return cy.task('saveSnapshot', {
+    file,
+    what: act,
+    exactSpecName,
+  }, { log: false })
+}
+
+const registerInCypress = () => {
+  // need to use correct sinon version for matcher.isMatcher to work
+  sinon = Cypress.sinon
+  const $ = Cypress.$
+
+  let snapshotIndex = {}
+
+  chai = window.chai
+  chai.Assertion.addMethod('matchDeep', matchDeepCypress)
+  chai.Assertion.addMethod('matchSnapshot', matchSnapshotCypress)
+
+  after(() => {
+    snapshotIndex = {}
+  })
+
+  before(() => {
+    addPluginButton($, 'toggle-snapshot-update', '', {
+      render () {
+        const btnIcon = $(this).children().first()
+
+        return btnIcon.text(Cypress.env('SNAPSHOT_UPDATE') ? 'snapshot\nupdate\non' : 'snapshot\nupdate\noff')
+        .css({ 'font-size': '10px', 'line-height': '0.9' })
+        .html(btnIcon.html().replace(/\n/g, '<br/>'))
+      },
+      click () {
+        const prev = Cypress.env('SNAPSHOT_UPDATE')
```

Comment:
don't use `Cypress.env`, just put this on a global since we intend to move away from the current `Cypress.env` implementation